### PR TITLE
ss-151: Add transaction table

### PR DESCRIPTION
**Pull request by Holovashcenko**
image
image

### CODE REVIEW

**Review by Evcat6 — APPROVED** [review]

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]
Can we make this size smaller?
image

**Comment by Holovashcenko** [comment]
image
image
image

**Comment by nikita-remeslov** [comment]
Conflicts

File: frontend/src/bundles/common/components/transanction-table/components/styles.module.scss [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+}
+
+.noTransaction {
+    justify-content: center;
+    margin-top: var(--m-4);
+    color: var(--color-blue-600);
+    font-weight: var(--font-weight-500);
+}
+
+@media (max-width: 576px) {
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/transanction-table/components/styles.module.scss:143]: media variable

File: frontend/src/bundles/common/components/transanction-table/components/future-transactions/future-transactions.tsx [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+            <div className={classNames(styles.totals, styles.transaction)}>
+                <div className={classNames(styles.columns)}>
+                    <FontAwesomeIcon
+                        className={styles.chevronIcon}
+                        icon={
+                            isShowFutureTransaction
+                                ? FaIcons.CHEVRON_DOWN
+                                : FaIcons.CHEVRON_UP
+                        }
+                        // eslint-disable-next-line react/jsx-no-bind
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/transanction-table/components/future-transactions/future-transactions.tsx:50]: ?

File: frontend/src/bundles/common/components/transanction-table/components/transaction/transaction.tsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    control,
+    errors,
+    isFutureTransaction = false,
+}) => {
+    const [isChecked, setIsChecked] = useState(false);
+
+    const handleChange = (event: React.FormEvent<HTMLFormElement>): void => {
+        const target = event.target as HTMLInputElement;
+        if (target.checked) {
+            // eslint-disable-next-line no-console
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/transanction-table/components/transaction/transaction.tsx:34]: ?

File: frontend/src/bundles/common/components/transanction-table/components/transaction/transaction.tsx [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    return (
+        <div
+            key={transaction.id}
+            className={classNames(
+                styles.transaction,
+                isChecked ? styles.checked : styles.unchecked,
+            )}
+        >
+            <div className={classNames(styles.columns, styles.leftColumn)}>
+                {/* eslint-disable-next-line react/jsx-no-bind  */}
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/transanction-table/components/transaction/transaction.tsx:51]: ?

File: frontend/src/bundles/common/components/transanction-table/helpers/group-transactions-by-date.helper.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    transactions: ITransaction[],
+): Record<string, ITransaction[]> => {
+    const groupedTransactions: Record<string, ITransaction[]> = {};
+
+    for (const transaction of dateToShortStringHelper(
+        transactions,
+    ) as ITransaction[]) {
+        const date = transaction.date;
+
+        // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/transanction-table/helpers/group-transactions-by-date.helper.ts:15]: ?

File: frontend/src/bundles/common/pages/dashboard/styles.module.scss [hunk 6]
```diff
@@ -12,7 +12,7 @@
 }
 
 .contentWrapper {
-    max-width: 1192px;
+    max-width: 1000px;
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/pages/dashboard/styles.module.scss:15]: I am not sure about this changes

Comment (nikita-remeslov) [reply frontend/src/bundles/common/pages/dashboard/styles.module.scss:15]: we also have variable for container size

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/common/components/transanction-table/components/future-transactions/future-transactions.tsx (hunk 2 above)
Comment (Holovashcenko) [reply frontend/src/bundles/common/components/transanction-table/components/future-transactions/future-transactions.tsx]: fix in [https://github.com/BinaryStudioAcademy/bsa-winter-2022-2023-smartspend/pull/192/commits/77b0aacb4b3e4c8404615f5670e0eaac4f165d78](https://github.com/BinaryStudioAcademy/bsa-winter-2022-2023-smartspend/pull/url)

File: frontend/src/bundles/common/components/transanction-table/components/styles.module.scss (hunk 1 above)
Comment (Holovashcenko) [reply frontend/src/bundles/common/components/transanction-table/components/styles.module.scss]: fix in [https://github.com/BinaryStudioAcademy/bsa-winter-2022-2023-smartspend/pull/192/commits/77b0aacb4b3e4c8404615f5670e0eaac4f165d78](https://github.com/BinaryStudioAcademy/bsa-winter-2022-2023-smartspend/pull/url)

**Comment by Holovashcenko** [comment]
image

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**